analyzer/codechecker_analyzer/analyzers/infer/analyzer.py

```diff
@@ -0,0 +1,240 @@
+
+        analyzer_cmd.extend(cmd_filtered)
+        LOG.debug_analyzer("Running analysis command "
+                           f"'{shlex.join(analyzer_cmd)}'")
+
+        return analyzer_cmd
+
+    @classmethod
+    def get_analyzer_checkers(cls):
+        """
+        Return the list of the supported checkers.
+        """
+        command = [cls.analyzer_binary(), "help", "--list-issue-types"]
+        desc = json.load(
+            open(Path(__file__).parent / "descriptions.json",
+                 "r", encoding="utf-8"))
+        checker_list = []
+        try:
+            output = subprocess.check_output(command,
+                                             stderr=subprocess.DEVNULL)
+            for entry in output.decode().split('\n'):
+                data = entry.strip().split(":")
+                if len(data) < 7:
+                    continue
+
+                entry_id = data[0].lower()
+                if entry_id in desc:
+                    description = desc[entry_id]
+                else:
+                    checker = data[6] if len(data) == 7 else data[5]
+                    description = f"used by '{checker}' checker"
+
+                entry_id = entry_id.replace("_", "-")
+                checker_list.append((f"infer-{entry_id}",
+                                     description))
+            return checker_list
+        except (subprocess.CalledProcessError) as e:
+            LOG.error(e.stderr)
+        except (OSError) as e:
+            LOG.error(e.errno)
+        return []
+
+    @classmethod
+    def get_analyzer_config(cls):
+        """
+        Config options for infer.
+        """
+        return []
+
+    @classmethod
+    def get_checker_config(cls):
+        """
+        Config options for infer checkers.
+        """
+        return []
+
+    def analyze(self, analyzer_cmd, res_handler, proc_callback=None):
+        result_handler = super().analyze(
+            analyzer_cmd, res_handler, proc_callback)
+
+        if result_handler.analyzer_stderr:
+            LOG.error(result_handler.analyzer_stderr)
+
+        return result_handler
+
+    def post_analyze(self, result_handler: InferResultHandler):
+        """
+        Post process the results after the analysis.
+        """
+
+    @classmethod
+    def resolve_missing_binary(cls, configured_binary, environ):
+        """
+        In case of the configured binary for the analyzer is not found in the
+        PATH, this method is used to find a callable binary.
+        """
+
+    @classmethod
+    def get_binary_version(cls, details=False) -> str:
+        """
+        Return the analyzer version.
+        """
+        # No need to LOG here, we will emit a warning later anyway.
+        if not cls.analyzer_binary():
+            return None
+        version = [cls.analyzer_binary(), '--version']
+        try:
+            output = subprocess.check_output(version,
+                                             env={'TZ': 'UTC'},
```

Comment:
Pelase initialize the environment like this https://github.com/Ericsson/codechecker/blob/321492ee2d2e25a0e2abc5910597678af535ba3f/analyzer/codechecker_analyzer/analyzers/cppcheck/analyzer.py#L90

and only extend the environment with TZ:UTC, not overwrite it.